lib/protoflow-blocks/src/types/hash_algorithm.rs

```diff
@@ -26,7 +39,119 @@ impl fmt::Display for HashAlgorithm {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         use HashAlgorithm::*;
         match self {
+            #[cfg(feature = "hash-blake3")]
             BLAKE3 => write!(f, "blake3"),
+            #[cfg(feature = "hash-sha2")]
+            SHA256 => write!(f, "sha256"),
+            #[cfg(feature = "hash-sha1")]
+            SHA1 => write!(f, "sha1"),
+            #[cfg(feature = "hash-md5")]
+            MD5 => write!(f, "md5"),
+        }
+    }
+}
+
+impl Default for HashAlgorithm {
+    fn default() -> Self {
+        #[cfg(feature = "hash-blake3")]
+        {
+            return HashAlgorithm::BLAKE3;
+        }
+        #[cfg(all(not(feature = "hash-blake3"), feature = "hash-sha2"))]
+        {
+            return HashAlgorithm::SHA256;
+        }
+        #[cfg(all(
+            not(feature = "hash-blake3"),
+            not(feature = "hash-sha2"),
+            feature = "hash-sha1"
+        ))]
+        {
+            return HashAlgorithm::SHA1;
+        }
+        #[cfg(all(
+            not(feature = "hash-blake3"),
+            not(feature = "hash-sha2"),
+            not(feature = "hash-sha1"),
+            feature = "hash-md5"
+        ))]
+        {
+            return HashAlgorithm::MD5;
+        }
+    }
+}
+
+pub trait Hasher {
+    fn compute_hash(&self, data: &[u8]) -> Vec<u8>;
+}
+
+#[cfg(feature = "hash-blake3")]
+struct Blake3 {}
+
+#[cfg(feature = "hash-blake3")]
+impl Hasher for Blake3 {
+    fn compute_hash(&self, data: &[u8]) -> Vec<u8> {
+        let mut hasher = blake3::Hasher::new();
+        hasher.update(data);
+        hasher.finalize().as_bytes().to_vec()
+    }
+}
+
+#[cfg(feature = "hash-sha2")]
+struct Sha256 {}
+
+#[cfg(feature = "hash-sha2")]
+impl Hasher for Sha256 {
+    fn compute_hash(&self, data: &[u8]) -> Vec<u8> {
+        use sha2::{Digest, Sha256};
+        let mut hasher = Sha256::new();
+        hasher.update(data);
+        hasher.finalize().to_vec()
+    }
+}
+
+#[cfg(feature = "hash-sha1")]
+struct Sha1 {}
+
+#[cfg(feature = "hash-sha1")]
+impl Hasher for Sha1 {
+    fn compute_hash(&self, data: &[u8]) -> Vec<u8> {
+        use sha1::{Digest, Sha1};
+        let mut hasher = Sha1::new();
+        hasher.update(data);
+        hasher.finalize().to_vec()
+    }
+}
+
+#[cfg(feature = "hash-md5")]
+struct Md5 {}
+
+#[cfg(feature = "hash-md5")]
+impl Hasher for Md5 {
+    fn compute_hash(&self, data: &[u8]) -> Vec<u8> {
+        use md5::{Digest, Md5};
+        let mut hasher = Md5::new();
+        hasher.update(data);
+        hasher.finalize().to_vec()
+    }
+}
+
+#[cfg(feature = "hash")]
+pub struct HasherFactory;
+
+#[cfg(feature = "hash")]
```

Comment:
Seems incorrect (and redundant) to enable this based on the "hash" feature. Shouldn't this whole file (Rust module) be behind any "hash-" feature being enabled?

Reply:
done